src/Libraries/RevitNodesUI/Selection.cs

```diff
@@ -49,16 +49,22 @@ public RevitDynamoModel RevitDynamoModel
             {
                 if (revitDynamoModel != null)
                 {
-                    var hwm = revitDynamoModel.Workspaces.OfType<HomeWorkspaceModel>().ElementAt(0);
-                    hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
+                    var hwm = revitDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+                    if (hwm != null)
+                    {
+                        hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
```

Comment:
Just curious why this is not being de-registered at the time of removing the workspace?